sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -196,9 +196,13 @@ object JdbcUtils extends Logging with SQLConfHelper {
     case java.sql.Types.CHAR => CharType(precision)
     case java.sql.Types.CLOB => StringType
     case java.sql.Types.DATE => DateType
-    case java.sql.Types.DECIMAL if precision != 0 || scale != 0 =>
-      DecimalType.bounded(precision, scale)
-    case java.sql.Types.DECIMAL => DecimalType.SYSTEM_DEFAULT
+    case java.sql.Types.DECIMAL | java.sql.Types.NUMERIC if precision != 0 || scale != 0 =>
+      if (scale < 0) {
+        DecimalPrecision.bounded(precision - scale, 0)
```

Comment:
Actually, I think these two changes are orthogonal. To support negative scale, we turn scale to 0, then it doesn't matter how we truncate the precision, as scale is already 0.

For non-negative scale, we can use a different way to truncate the precision. It's better to do it in a separate PR as it's a behavior change.

Reply:
+1, make sense